optimism/src/lookup.rs

```diff
@@ -0,0 +1,116 @@
+use ark_ff::{Field, One};
+
+#[derive(Copy, Clone, Debug)]
+pub enum Sign {
+    Pos,
+    Neg,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum LookupMode {
+    Read,
+    Write,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum LookupTable {
+    MemoryLookup,
+    RegisterLookup,
+    // Single-column table of 2^16 entries with the sparse representation of all values
+    SparseLookup,
+    // Single-column table of all values in the range [0, 2^16)
+    RangeCheck16Lookup,
+    // Dual-column table of all values in the range [0, 2^16) and their sparse representation
+    ResetLookup,
+    // 24-row table with all possible values for round and their round constant in expanded form
+    RoundConstantsLookup,
+    // All [0..136] values of possible padding lengths, the value 2^len, and the 5 corresponding pad suffixes with the 10*1 rule
+    PadLookup,
+    // All values that can be stored in a byte (amortized table, better than model as RangeCheck16 (x and scaled x)
+    ByteLookup,
+    // Input/Output of Keccak steps
+    KeccakStepLookup,
+    // Syscalls communication channel
+    SyscallLookup,
+}
+
+#[derive(Clone, Debug)]
+pub struct Lookup<Fp> {
+    pub mode: LookupMode,
+    /// The number of times that this lookup value should be added to / subtracted from the lookup accumulator.    pub magnitude_contribution: Fp,
```

Comment:
I suppose the field can be removed?

Reply:
Yeah probably, it's always 1 in our case but it is more general if we include it.